pkg/controller/disruption/disruption.go

```diff
@@ -814,9 +835,44 @@ func (dc *DisruptionController) updatePdbStatus(pdb *policy.PodDisruptionBudget,
 		ObservedGeneration: pdb.Generation,
 	}
 
+	UpdateDisruptionAllowedCondition(newPdb)
+
 	return dc.getUpdater()(newPdb)
 }
 
+func UpdateDisruptionAllowedCondition(pdb *policy.PodDisruptionBudget) {
```

Comment:
this is used by the API server (in eviction storage) and by controller manager. suggest putting this in `staging/src/k8s.io/component-helpers/apps/poddisruptionbudget` so `pkg/registry` doesn't take a dependency on `pkg/controller`